pkg/api/customization/cluster/validator.go

```diff
@@ -132,13 +132,23 @@ func (v *Validator) checkClusterForEnforcement(spec *mgmtclient.Cluster) bool {
 func (v *Validator) isTemplateEnforced(spec *mgmtclient.Cluster) (bool, error) {
 	//template must be marked as enforced=true  + must be public
 
-	split := strings.SplitN(spec.ClusterTemplateID, ":", 2)
+	split := strings.SplitN(spec.ClusterTemplateRevisionID, ":", 2)
 	if len(split) != 2 {
-		return false, fmt.Errorf("error in splitting clusterTemplate name %v", spec.ClusterTemplateID)
+		return false, fmt.Errorf("error in splitting clusterTemplateRevision name %v", spec.ClusterTemplateRevisionID)
+	}
+	revName := split[1]
+	clusterTempRev, err := v.ClusterTemplateRevisionLister.Get(namespace.GlobalNamespace, revName)
+	if err != nil {
+		return false, err
 	}
-	templateName := split[1]
 
-	clusterTemp, err := v.ClusterTemplateLister.Get(namespace.GlobalNamespace, templateName)
+	templateIDStr := clusterTempRev.Spec.ClusterTemplateName
+	split = strings.Split(templateIDStr, ":")
+	if len(split) != 2 {
+		return false, fmt.Errorf("error in splitting clusterTemplate name %v", templateIDStr)
```

Comment:
will it be a template revision? then we'd have to change the error message

Reply:
no, it is clusterTemplate